### PR TITLE
eof: fix multi-container-section case

### PR DESCRIPTION
**Pull request by pdobacz**
Credit to @chfast to find the case via fuzzing, as well as suggest this version of the fix (alternative version in comment, not sure which is cleaner anymore)

### CODE REVIEW

File: lib/evmone/eof.cpp [hunk 1]
```diff
@@ -106,7 +107,7 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_section_headers(byt
             // If DATA_SECTION is expected, CONTAINER_SECTION is also allowed, because
             // container section is optional.
             if (section_id != expected_section_id &&
-                (expected_section_id != DATA_SECTION || section_id != CONTAINER_SECTION))
+                (expected_section_id != CONTAINER_SECTION || section_id != DATA_SECTION))
```
Comment (pdobacz) [inline lib/evmone/eof.cpp:110]: alternative is to change the condition to

```
            if (section_id != expected_section_id &&
                !(expected_section_id == DATA_SECTION && section_id == CONTAINER_SECTION &&
                    section_headers[CONTAINER_SECTION].size() == 0))
```

and leave the rest as is

**Comment by chfast** [comment]
Let me try one more thing here.

**Comment by chfast** [comment]
I pushed an alternative fix where we check `id != expected` twice but the helper `get_section_missing_error()` remains sane.

File: lib/evmone/eof.cpp [hunk 2]
```diff
@@ -104,10 +103,11 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_section_headers(byt
         {
             section_id = *it++;
 
-            // If DATA_SECTION is expected, CONTAINER_SECTION is also allowed, because
-            // container section is optional.
-            if (section_id != expected_section_id &&
-                (expected_section_id != CONTAINER_SECTION || section_id != DATA_SECTION))
+            // Skip optional sections.
+            if (section_id != expected_section_id && expected_section_id == CONTAINER_SECTION)
```
Comment (pdobacz) [inline lib/evmone/eof.cpp:107]: matter of taste, but this reads weirder to me, weirder than the `get_section_missing_error` modification, and weirder than the "2nd" alternative which checks `CONTAINER_SECTION.size()` (which also keeps `get_section_missing_error` intact). The weirdness comes from the fact that the `expected_section_id` isn't a check-modify, but a check-maybe modify-check-modify sequence on each iteration.

Maybe we need a 3rd opinion to pick the least obtrusive fix, @gumb0 ?

Comment (gumb0) [reply lib/evmone/eof.cpp:107]: I think definitely this one here is more understandable.

**Review by gumb0 — APPROVED** [review]

**Comment by chfast** [comment]
This issue was found indirectly by fuzzing: I was debugging the fuzzer mutation and noticed this weird EOF header. This indicates this validation issue may not be exploitable. Yet, it is a serious bug.

**Pull request merged**